src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -125,10 +132,15 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseRep
             throw new BadRequestAlertException("A complaint can be only associated to a result", COMPLAINT_ENTITY_NAME, "noresultid");
         }
 
-        if (complaintService.getByResultId(complaint.getResult().getId()).isPresent()) {
+        if (complaintRepository.findByResultId(complaint.getResult().getId()).isPresent()) {
             throw new BadRequestAlertException("A complaint for this result already exists", COMPLAINT_ENTITY_NAME, "complaintexists");
         }
 
+        Result result = resultRepository.findOneElseThrow(complaint.getResult().getId());
+        if (!authCheckService.isOwnerOfParticipation((StudentParticipation) result.getParticipation())) {
```

Comment:
Could we add a method isOwnerOfParticipationOrThrow which automatically throws a forbidden exception in case the user is not the owner of the participation? We use a similar approach for regular authorization checks